Frameworks/Foundation/NSOperationQueue.mm

```diff
@@ -200,15 +353,11 @@ - (NSOperationQueue*)_doMainWork {
         didWork = FALSE;
 
         for (int i = 0; i < NSOperationQueuePriority_Count; i++) {
-            [priv->suspendedCondition lock];
-            while (priv->isSuspended) {
-                [priv->suspendedCondition wait];
-            }
-            [priv->suspendedCondition unlock];
-
-            if (RunOperationFromLists(&priv->myQueues[i], &priv->queues[i], &priv->curOperation)) {
-                didWork = TRUE;
-            }
+            // Wait until the operation queue is no longer suspended.
+            // NOTE: if an operation queue becomes suspended after this point, the "in flight" operation
+            // will not be suspended, it will continue as if the queue is still running.
+            _priv->WaitUntilNotSuspended();
+            didWork |= _priv->RunOperationFromLists(i);
```

Comment:
Although that could change the behavior because of short circuiting.

---
In reply to: [95450897](https://github.com/Microsoft/WinObjC/pull/1676#discussion_r95450897) [](ancestors = 95450897)